src/Data/Vec/Functional/Algebra/Base.agda

```diff
@@ -0,0 +1,53 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some Vector-related module Definitions
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Function using (_$_)
+open import Data.Product hiding (map)
+open import Data.Nat using (ℕ)
+open import Data.Fin using (Fin)
+open import Data.Vec.Functional
+open import Algebra.Core
+open import Algebra.Bundles
+open import Algebra.Module
+open import Relation.Binary
+import Data.Vec.Functional.Relation.Binary.Equality.Setoid as VecSetoid
+import Algebra.Definitions as AD
+import Algebra.Structures as AS
+
+module Data.Vec.Functional.Algebra.Base
+  {c ℓ} (cring : CommutativeRing c ℓ) where
```

Comment:
I'm not happy about this whole module being parameterized on `CommutativeRing`, as many of the definitions work with smaller algebraic structures, but I'm not sure what to do about it. Maybe other people have some ideas?